tools/clang/lib/Sema/SemaExpr.cpp

```diff
@@ -4117,6 +4117,16 @@ Sema::ActOnArraySubscriptExpr(Scope *S, Expr *base, SourceLocation lbLoc,
     idx = result.get();
   }
 
+  // HLSL Change Starts - Check for subscript access of out indices
+  // Disallow component access for out indices for DXIL path. We still allow
+  // this in SPIR-V path.
+  if (!getLangOpts().SPIRV && base->getType()->isRecordType() &&
+      IsExprAccessingMeshOutArray(base)) {
```

Comment:
I think `IsExprAccessingMeshOutArray` is overly inclusive.  It should only apply to the index array, but this would apply to vertex and primitive arrays as well.

Reply:
Also, this is in the context of regular clang code, so I think we need to check `getLangOpts().HLSL && !getLangOpts().SPIRV`.

Reply:
Done!